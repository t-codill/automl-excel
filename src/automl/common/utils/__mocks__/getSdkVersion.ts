export const getSdkVersion = () => "1.0.0.*";
