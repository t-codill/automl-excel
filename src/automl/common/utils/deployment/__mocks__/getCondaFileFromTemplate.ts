import { RunHistoryAPIsModels } from "@vienna/runhistory";

export const getCondaFileFromTemplate = (
    run: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDetailsDto | undefined
): string | undefined => {
    if (!run) {
        return "";
    }

    return `mock conda file content for run: ${run.runId}`;
};
