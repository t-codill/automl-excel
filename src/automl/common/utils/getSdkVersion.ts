import { RunHistoryAPIsModels } from "@vienna/runhistory";
import { defaultSdkVersion } from "../defaultSdkVersion";
import { safeParseJson } from "./safeParseJson";

export const getSdkVersion = (
    run: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDetailsDto | undefined,
    defaultVersion: string = defaultSdkVersion
) => {
    if (!run || !run.properties || !run.properties.dependencies_versions) {
        return defaultVersion;
    }
    const dependenciesVersions = safeParseJson(run.properties.dependencies_versions);
    if (!dependenciesVersions
        || !dependenciesVersions["azureml-train-automl"]) {
        return defaultVersion;
    }
    return dependenciesVersions["azureml-train-automl"];
};
