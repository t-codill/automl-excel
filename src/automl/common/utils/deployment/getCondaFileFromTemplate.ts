import { RunHistoryAPIsModels } from "@vienna/runhistory";
import { getSdkVersion } from "../getSdkVersion";

const condaEnvTemplate = `
# Conda environment specification. The dependencies defined in this file will
# be automatically provisioned for runs with userManagedDependencies=False.
# Details about the Conda environment file format:
# https://conda.io/docs/user-guide/tasks/manage-environments.html#create-env-file-manually

name: project_environment
dependencies:
  # The python interpreter version.
  # Currently Azure ML only supports 3.5.2 and later.
- python=3.6.2

- pip:
  - azureml-train-automl==##sdkVersion##
- numpy
- scikit-learn
- py-xgboost<=0.80
`;

export const getCondaFileFromTemplate = (
  run: RunHistoryAPIsModels.MicrosoftMachineLearningRunHistoryContractsRunDetailsDto | undefined
): string => {
  const sdkVersion = getSdkVersion(run);
  return condaEnvTemplate.replace("##sdkVersion##", sdkVersion);
};
