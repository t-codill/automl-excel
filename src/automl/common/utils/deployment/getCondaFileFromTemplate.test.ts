import * as getSdkVersionModule from "../getSdkVersion";
import { getCondaFileFromTemplate } from "./getCondaFileFromTemplate";

jest.mock("../getSdkVersion");

describe("getCondaFileFromTemplate", () => {
    it("should return undefined if experiment name is undefined", () => {
        expect(getCondaFileFromTemplate({}))
            .toMatchSnapshot();
    });

    it("should call getSdkVersion with expected run", () => {
        const sp = jest.spyOn(getSdkVersionModule, "getSdkVersion");

        expect(getCondaFileFromTemplate({
            runId: "foo"
        }));

        expect(sp)
            .toMatchSnapshot();
    });
});
