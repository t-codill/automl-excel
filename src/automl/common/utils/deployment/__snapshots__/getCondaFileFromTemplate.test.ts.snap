// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCondaFileFromTemplate should call getSdkVersion with expected run 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "runId": "foo",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": "1.0.0.*",
    },
  ],
}
`;

exports[`getCondaFileFromTemplate should return undefined if experiment name is undefined 1`] = `
"
# Conda environment specification. The dependencies defined in this file will
# be automatically provisioned for runs with userManagedDependencies=False.
# Details about the Conda environment file format:
# https://conda.io/docs/user-guide/tasks/manage-environments.html#create-env-file-manually

name: project_environment
dependencies:
  # The python interpreter version.
  # Currently Azure ML only supports 3.5.2 and later.
- python=3.6.2

- pip:
  - azureml-train-automl==1.0.0.*
- numpy
- scikit-learn
- py-xgboost<=0.80
"
`;
